hs_core/views/__init__.py

```diff
@@ -261,27 +258,61 @@ class AddGroupForm(forms.Form):
         group = forms.ModelChoiceField(Group.objects.all(), widget=autocomplete_light.ChoiceWidget("GroupAutocomplete"))
 
 
-    res, _, _ = authorize(request, shortkey, edit=True, full=True, superuser=True)
+    res, _, user = authorize(request, shortkey, edit=True, full=True, superuser=True)
     t = request.POST['t']
     values = [int(k) for k in request.POST.getlist('designees', [])]
     if t == 'owners':
-        res.owners = User.objects.in_bulk(values)
+        owners_to_keep = User.objects.in_bulk(values).values()
+        for owner in res.raccess.edit_users.all():
+            if owner not in owners_to_keep:
+                try:
+                    user.uaccess.unshare_resource_with_user(res, owner)
+                except HSAccessException:
+                    #TODO: propogate this error to the user interface and leave the loop
+                    pass
```

Comment:
You remove users but never add the new ones.

Also is `edit_users` really "owners"?